`torch.compile` caching of config fields should be opt-out by default

nadathurv opened the pull request:
## Opt-out hashing for torch.compile cache keys (ModelConfig + envs)

Fixes [#23107](https://github.com/vllm-project/vllm/issues/23107). Implements the RFC in [#16501](https://github.com/vllm-project/vllm/issues/16501)

check failures are due to the opt-out hashing working.

ProExpertProg commented on vllm/config/__init__.py:
```diff
@@ -483,26 +483,93 @@ def compute_hash(self) -> str:
+
+        # Normalize common types; skip opaque objects to avoid
+        # identity-dependent or process-local representations.
+        def _canon(x):
+            if x is None or isinstance(x, (bool, int, float, str)):
+                return x
+            if isinstance(x, enum.Enum):
+                return x.value
+            try:
+                import torch
+                if isinstance(x, torch.dtype):
+                    return str(x)
+            except Exception:
+                pass
+            if isinstance(x, pathlib.Path):
+                return str(x)
+            if isinstance(x, dict):
+                return tuple(sorted((str(k), _canon(v)) for k, v in x.items()))
+            if isinstance(x, set):
+                return tuple(sorted(repr(_canon(v)) for v in x))
+            if isinstance(x, (list, tuple)):
+                return tuple(_canon(v) for v in x)
+            raise TypeError
```
This repeats in all modified files, can you extract it to a common place?

nadathurv replied:
Done. I moved this to utils.py.

ProExpertProg reviewed (commented):
A few comments. We definitely need as many people's eyes on this as possible to make sure we get the exclude lists right. cc @zou3519 @youkaichao

nadathurv commented on vllm/config/parallel.py:
```diff
@@ -226,19 +226,54 @@ def sync_kv_cache_memory_size(dp_group: ProcessGroup,
+
+        EXCLUDE_FROM_HASH = {
+            # Derived/runtime topology, networking, or launch details
+            "world_size",
```
Fixing it right now. Will have most of the feedback up in the next commit.

Commit:
Feedback Addressed

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
incorporating feedback

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
improving utils

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
opt-out hashing: env hash helper + stricter HF config hashing; unify field iteration via utils; tests

Commit:
Merge branch 'opt-out-caching-tc' of https://github.com/nadathurv/vllm into opt-out-caching-tc

Commit:
Delete test

Commit:
refactor & more feedback implemenation

Added block size
Fixed linter, long line docustring, comments, indentations, namings, etc.,

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
updates

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

bnellnm commented on vllm/config/parallel.py:
```diff
@@ -266,19 +270,43 @@ def sync_kv_cache_memory_size(dp_group: ProcessGroup,
+            "worker_cls",
+            "sd_worker_cls",
+            "worker_extension_cls",
+            "enable_multimodal_encoder_data_parallel",
```
I don't think this should be excluded.  It appears to be used to check whether or not data parallelism should be enabled in a few models.

Commit:
updates

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

bnellnm commented on vllm/config/utils.py:
```diff
@@ -18,12 +18,137 @@ def config(cls: ConfigT) -> ConfigT:
+    if isinstance(x, Sequence) and not isinstance(x, (str, bytes, bytearray)):
+        return tuple(canon_value(v) for v in x)
+
+    # Unsupported type
```
Can we have some kind of debug logging for unsupported types?  It seems like if an unsupported member type is encountered, it silently skips including it in the hash?  I feel like this could lead to subtle/hard to find bugs.

nadathurv replied:
Fixed in most recent commit

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Commit:
addressing feedback

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'opt-out-caching-tc' of https://github.com/nadathurv/vllm into opt-out-caching-tc

nadathurv commented on vllm/config/compilation.py:
```diff
@@ -352,16 +350,33 @@ def compute_hash(self) -> str:
+            "local_cache_dir",
+            "bs_to_padded_graph_size",
+            "enabled_custom_ops",
+            "disabled_custom_ops",
```
Fixed!

Commit:
pre-commit fixes

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

bnellnm commented on vllm/config/utils.py:
```diff
@@ -79,6 +79,12 @@ def canon_value(x):
 
     # Unsupported type
+    try:
+        import logging
```
Is there some reason the vllm logger doesn't work here?

Commit:
implemented vllm logger

feeeeedbaaack

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
update

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
fixes

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
fixes2

implementing feedback

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update cache.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update compilation.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update parallel.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update utils.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update envs.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
fix pre-commit

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
implementing feedback

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

nadathurv commented on vllm/config/__init__.py:
```diff
@@ -2483,16 +2508,19 @@ def compute_hash(self) -> str:
+
+        items = _build_items(self, EXCLUDE_FROM_HASH)
+
+        hash_str = hashlib.md5(repr(tuple(items)).encode(),
```
Done!

nadathurv commented on vllm/config/utils.py:
```diff
@@ -80,11 +80,17 @@ def canon_value(x):
-                                          type(x).__name__)
+        from vllm.logger import init_logger
+        init_logger(__name__).debug("canon_value: unsupported type '%s'",
+                                    type(x).__name__)
```
Done. Thanks!

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update envs.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update envs.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

bnellnm commented on vllm/config/utils.py:
```diff
@@ -27,3 +31,153 @@ def config(cls: ConfigT) -> ConfigT:
+    import logging
+    from contextlib import suppress
+
+    logger = logging.getLogger(__name__)
```
Use global `logger`

Commit:
global logger

implementing feedback
Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
utils.py update

implementing feedback.
Signed-off-by: nadathurv <work.vnadathur@gmail.com>

nadathurv commented on vllm/config/utils.py:
```diff
@@ -27,3 +31,142 @@ def config(cls: ConfigT) -> ConfigT:
+    import enum
+    import pathlib
+    from collections.abc import Mapping, Sequence, Set
+    from contextlib import suppress
```
Fixed!

Commit:
Update __init__.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

nadathurv commented on vllm/config/__init__.py:
```diff
@@ -514,26 +514,51 @@ def compute_hash(self) -> str:
+        # graph or are unstable. See RFC #16501.
+        MODEL_EXCLUDE_FROM_HASH = {
+            "tokenizer",
+            "hf_config",  # hash content via JSON below
```
tokenizer and hf_config aren't special case in other configs. If it is relevant in another config, handling in build_opt_out_items risks under-hashing. 

If you meant having the exclude lists as part of utils, it's not possible given that they are specific to the configs.

I feel like I don't fully understand what is desired here. Could you elaborate?

hmellor replied:
Sorry you're right, `hf_config` is only a member of `ModelConfig`

Commit:
remaing the exclude from hash

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

hmellor commented on vllm/config/__init__.py:
```diff
@@ -493,26 +493,72 @@ def compute_hash(self) -> str:
+        MODEL_EXCLUDE_FROM_HASH = {
+            "enforce_eager",
+            "tokenizer",
+            "hf_config",  # hash content via JSON below
```
Do you mean specifically the `hf_config` or all the fields of `ModelConfig`?

hmellor commented on vllm/config/cache.py:
```diff
@@ -131,13 +131,32 @@ def compute_hash(self) -> str:
+            "kv_sharing_fast_prefill",
+        }
+
+        from vllm.config.utils import build_opt_out_items as _build_items
```
Could we also remove the renaming of the imports in the other files too?

nadathurv replied:
done!

Commit:
fixed import & naming cases

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Update vllm/config/__init__.py

Co-authored-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: nadathurv <work.vnadathur@gmail.com>

nadathurv commented on vllm/config/utils.py:
```diff
@@ -27,3 +36,129 @@ def config(cls: ConfigT) -> ConfigT:
+    - Includes declared fields not in `exclude`.
+    - Skips non-canonicalizable values.
+    """
+    items: list[tuple[str, object]] = []
```
I didn't want to use a dict as a return type because the callers get a list of (key, value) tuples, then append other tuples (env factors or hf_config content), and then hash or tuple directly. 

A dict would break those usages or force churn across multiple configs and the hasher. Also the hashing relies on a stable order (repr(tuple(items))). With a dict it would need to re-sort and re-serialize at each call. 

We could build a dict locally and then return a list of tuples but it adds nothing functionally because field names are unique and we already iterate in a sorted order. Plus, keeping a list of tuples also stays consistent with `build_opt_items_override` across configs.

Let me know how you want me to proceed.

hmellor replied:
Since this is a new function, you could make the callers expect a dict. I ask because a list of (key, value) tuples is effectively a dict.

> With a dict it would need to re-sort and re-serialize at each call.

Dicts have been ordered since Python 3.6, we would not need to resort them.

nadathurv replied:
Got it. I understand. Tysm!

Commit:
Merge branch 'main' into opt-out-caching-tc

hmellor committed:
Simplify

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

hmellor committed:
fix factors type

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

ProExpertProg reviewed (commented):
A few minor notes, this is starting to look very clean!

Commit:
Warnings and byte handling

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

zou3519 commented on vllm/config/utils.py:
```diff
@@ -27,3 +38,103 @@ def config(cls: ConfigT) -> ConfigT:
+    if x is None or isinstance(x, (bool, int, float, str)):
+        return x
+
+    # Enums by value
+    if isinstance(x, enum.Enum):
+        return x.value
```
x.value is an int?
Should do normalize_value(x.value) instead (I'm not completely sure it's always an int)

zou3519 commented on vllm/config/utils.py:
```diff
@@ -27,3 +38,103 @@ def config(cls: ConfigT) -> ConfigT:
+    # Callable by qualified name
+    try:
+        if callable(x):
+            module = getattr(x, "__module__", "")
+            qual = getattr(x, "__qualname__", repr(x))
+            return f"{module}.{qual}" if module else qual
```
This won't work with lambdas.  
```
m = lambda x: x
print(m.__qualname__)
# "<lambda>"
```

zou3519 commented on vllm/config/utils.py:
```diff
@@ -27,3 +38,103 @@ def config(cls: ConfigT) -> ConfigT:
+    with suppress(Exception):
+        logger.debug("normalize_value: unsupported type '%s'",
+                     type(x).__name__)
+    raise TypeError
```
TypeError("msg") ?

zou3519 commented on vllm/config/utils.py:
```diff
@@ -27,3 +38,103 @@ def config(cls: ConfigT) -> ConfigT:
+        return x.to_json_string()
+
+    # Unsupported type
+    with suppress(Exception):
+        logger.debug("normalize_value: unsupported type '%s'",
+                     type(x).__name__)
```
Why suppress Exception?

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Update utils.py

implementing feedback
Signed-off-by: nadathurv <work.vnadathur@gmail.com>

nadathurv commented on vllm/config/utils.py:
```diff
@@ -27,3 +38,103 @@ def config(cls: ConfigT) -> ConfigT:
+            qual = getattr(x, "__qualname__", repr(x))
+            return f"{module}.{qual}" if module else qual
+    except Exception:
+        pass
```
`logits_processors: Optional[list[Union[str, type[LogitsProcessor]]]] = None` and `hf_overrides` in ModelConfig are both examples of callable that are handled by this.

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
fix callable handling

implementing feedback
Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
small improvements

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

zou3519 commented on vllm/config/utils.py:
```diff
@@ -18,12 +24,113 @@ def config(cls: ConfigT) -> ConfigT:
     return cls
+
+
+def normalize_value(x):
```
One Edge Case is that Enum with value 1 would map to the same value as int with value 1. So we would really require that those two things are different w.r.t. to the hash (a config should not do something different with Enum(value=1) and int(1).

The more robust way to do this is probably for normalize_value(x) to return (type_str, value) or something. So we end up getting ("foo.Enum", 1) and ("int", 1) separately.

nadathurv replied:
added

Commit:
update utils

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

nadathurv commented on vllm/config/utils.py:
```diff
@@ -18,12 +24,113 @@ def config(cls: ConfigT) -> ConfigT:
+        return normalize_value(x.value)
+
+    # Callables: classes, functions (incl. lambdas/nested), callable instances
+    if callable(x):
```
in the latest commit, i removed support for nested functions and lambdas per your feedback. my unit test passes. I can paste it if helpful.

Commit:
Merge branch 'main' into opt-out-caching-tc

Commit:
Merge branch 'main' into opt-out-caching-tc

zou3519 commented on vllm/config/utils.py:
```diff
@@ -18,12 +24,104 @@ def config(cls: ConfigT) -> ConfigT:
+        enum_type = f"{x.__class__.__module__}.{x.__class__.__qualname__}"
+        return (enum_type, normalize_value(x.value))
+
+    # Callables: allow classes (types) only; reject functions/lambdas/methods
+    if isinstance(x, type):
+        module = getattr(x, "__module__", "")
+        qual = getattr(x, "__qualname__", getattr(x, "__name__", ""))
+        return ".".join([p for p in (module, qual) if p]) or repr(x)
+    if callable(x):
+        raise TypeError(
+            "normalize_value: function or callable instance unsupported")
```
What is an example of a class that we allow? If the thing is an instance of a class, then it's actually an object with state on it.

Commit:
Update envs.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update utils.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

ProExpertProg commented on vllm/config/utils.py:
```diff
@@ -18,12 +24,104 @@ def config(cls: ConfigT) -> ConfigT:
     return cls
+
+
+def normalize_value(x):
```
++

WorldExplored committed:
Created Unit Test

test checks normalize value functn. verifies hashing and consistent behavior.

Signed-off-by: <Srreyansh Sethi srreyansh.sethi@gmail.com>
Co-Authored-By: nadathurv <work.vnadathur@gmail.com>

WorldExplored committed:
Refactor Comments

I refactored some comments

WorldExplored committed:
Pre-comit fix

Signed-off-by: <Srreyansh Sethi srreyansh.sethi@gmail.com>

WorldExplored committed:
Fixed Precomit Error

Signed-off-by: <Srreyansh Sethi srreyansh.sethi@gmail.com>

WorldExplored committed:
Fixed pre-commit issue

Signed-off-by: <Srreyansh Sethi srreyansh.sethi@gmail.com>

WorldExplored committed:
Fixing Precommit

Signed-off-by: <Srreyansh Sethi srreyansh.sethi@gmail.com>

hmellor commented:
You can fix pre-commit in one go by installing it locally

```
uv pip install pre-commit
pre-commit install
pre-commit run -a
```

Please also fix DCO

WorldExplored committed:
Precomit Fix

Fixed code imports

Signed-off-by: <Srreyansh Sethi srreyansh.sethi@gmail.com>

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Commit:
updating init.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update __init__.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update utils.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update utils.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update utils.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
fix var

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

WorldExplored committed:
updated cache

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

nadathurv commented:
@ProExpertProg @zou3519 @hmellor The normalized factors are now added to the cache. Can review of this pr continue?

Commit:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update utils.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
moving opt-out to modelconfig file

modelconfig was moved to model.py, so I also moved the opt-out focused hashing

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update __init__.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update __init__.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Update model.py

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

Commit:
Merge branch 'opt-out-caching-tc' of https://github.com/nadathurv/vllm into opt-out-caching-tc

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Update parallel.py

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

ProExpertProg commented on vllm/compilation/backends.py:
```diff
@@ -551,6 +551,103 @@ def __call__(self, graph: fx.GraphModule, example_inputs) -> Callable:
+            env_factors = {}
+
+        try:
+            config_factors: dict[str, object] = {}
```
Instead of this, why don't we convert `compute_hash`:

```
# old
def compute_hash() -> str:

# new
def compile_factors() -> dict[str, ...]:
```

So that we can extract the relevant compile factors in raw format (for logging) and then also hash them when needed.

WorldExplored committed:
adjusted envs and backends.

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

ProExpertProg commented on vllm/envs.py:
```diff
@@ -1487,82 +1486,39 @@ def set_vllm_use_v1(use_v1: bool):
-
-    return hash_str
+    ignored_factors: set[str] = {
+        "MAX_JOBS", "VLLM_RPC_BASE_PATH", "VLLM_USE_MODELSCOPE",
```
I think the Marlin ones shouldn't be ignored? cc @LucasWilkinson

ProExpertProg commented on vllm/envs.py:
```diff
@@ -1470,48 +1470,39 @@ def set_vllm_use_v1(use_v1: bool):
-            except Exception:
-                pass
+            # Do not drop the factor; mark retrieval failure deterministically.
+            factors[factor] = "<error:unavailable>"
```
I'm not a huge fan of these, `getter()` should never fail.

ProExpertProg commented on vllm/envs.py:
```diff
@@ -1470,48 +1470,39 @@ def set_vllm_use_v1(use_v1: bool):
+        except Exception:
+            # Preserve the factor with a stable placeholder to avoid
+            # under-hashing.
+            factors[factor] = f"<unserializable:{type(raw).__name__}>"
```
Why are we ignoring an error in `normalize_value` here? I think it would be better to fail here so that whoever added an unserializable factor can resolve it.

ProExpertProg commented on vllm/compilation/backends.py:
```diff
@@ -485,7 +487,8 @@ def __call__(self, graph: fx.GraphModule, example_inputs) -> Callable:
             # VLLM_PP_LAYER_PARTITION will affect the computation graph.
-            env_hash = envs.compute_hash()
+            from vllm.config.utils import hash_factors
+            env_hash = hash_factors(envs.compile_factors())
```
This way you can reuse `env_factors` below?
```suggestion
            env_factors = envs.compile_factors()
            env_hash = hash_factors(env_factors)
```

ProExpertProg reviewed (changes requested):
Still way too much improper error handling - this needs to be robust and perhaps we can catch errors and rethrow them with helpful error messages or log them etc. but we shouldn't just move on!

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
refactored envs and backends

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>
Co-Authored-By: nadathurv <work.vnadathur@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

Commit:
Merge branch 'vllm-project:main' into opt-out-caching-tc

Commit:
import fix

Signed-off-by: nadathurv <work.vnadathur@gmail.com>

WorldExplored committed:
Adjusted logger logic and utils imports

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

Commit:
Merge branch 'main' into opt-out-caching-tc

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

WorldExplored committed:
Update model.py for precommit

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Fixed precommit

Signed-off-by: WorldExplored <srreyansh.sethi@gmail.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>

WorldExplored committed:
Merge branch 'main' into opt-out-caching-tc

Signed-off-by: Srreyansh Sethi <107075589+WorldExplored@users.noreply.github.com>
